Traceback (most recent call last):
  File "/opt/anaconda3/envs/causal-inference-book/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/anaconda3/envs/causal-inference-book/lib/python3.11/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/causal-inference-book/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 173, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/causal-inference-book/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/causal-inference-book/lib/python3.11/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/opt/anaconda3/envs/causal-inference-book/lib/python3.11/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/anaconda3/envs/causal-inference-book/lib/python3.11/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
import matplotlib.pyplot as plt

# Plot the posterior distributions
x = np.linspace(0, 1, 100)
for i in range(len(arms)):
    plt.plot(x, np.random.beta(alpha[i], beta[i], size=1000), label=f'Arm {arms[i]}')

plt.title('Posterior Distributions of Arms')
plt.xlabel('Reward Probability')
plt.ylabel('Density')
plt.legend()
plt.show()

------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mValueError[0m                                Traceback (most recent call last)
Cell [0;32mIn[3], line 6[0m
[1;32m      4[0m x [38;5;241m=[39m np[38;5;241m.[39mlinspace([38;5;241m0[39m, [38;5;241m1[39m, [38;5;241m100[39m)
[1;32m      5[0m [38;5;28;01mfor[39;00m i [38;5;129;01min[39;00m [38;5;28mrange[39m([38;5;28mlen[39m(arms)):
[0;32m----> 6[0m     plt[38;5;241m.[39mplot(x, np[38;5;241m.[39mrandom[38;5;241m.[39mbeta(alpha[i], beta[i], size[38;5;241m=[39m[38;5;241m1000[39m), label[38;5;241m=[39m[38;5;124mf[39m[38;5;124m'[39m[38;5;124mArm [39m[38;5;132;01m{[39;00marms[i][38;5;132;01m}[39;00m[38;5;124m'[39m)
[1;32m      8[0m plt[38;5;241m.[39mtitle([38;5;124m'[39m[38;5;124mPosterior Distributions of Arms[39m[38;5;124m'[39m)
[1;32m      9[0m plt[38;5;241m.[39mxlabel([38;5;124m'[39m[38;5;124mReward Probability[39m[38;5;124m'[39m)

File [0;32m/opt/anaconda3/envs/causal-inference-book/lib/python3.11/site-packages/matplotlib/pyplot.py:3590[0m, in [0;36mplot[0;34m(scalex, scaley, data, *args, **kwargs)[0m
[1;32m   3582[0m [38;5;129m@_copy_docstring_and_deprecators[39m(Axes[38;5;241m.[39mplot)
[1;32m   3583[0m [38;5;28;01mdef[39;00m [38;5;21mplot[39m(
[1;32m   3584[0m     [38;5;241m*[39margs: [38;5;28mfloat[39m [38;5;241m|[39m ArrayLike [38;5;241m|[39m [38;5;28mstr[39m,
[0;32m   (...)[0m
[1;32m   3588[0m     [38;5;241m*[39m[38;5;241m*[39mkwargs,
[1;32m   3589[0m ) [38;5;241m-[39m[38;5;241m>[39m [38;5;28mlist[39m[Line2D]:
[0;32m-> 3590[0m     [38;5;28;01mreturn[39;00m gca()[38;5;241m.[39mplot(
[1;32m   3591[0m         [38;5;241m*[39margs,
[1;32m   3592[0m         scalex[38;5;241m=[39mscalex,
[1;32m   3593[0m         scaley[38;5;241m=[39mscaley,
[1;32m   3594[0m         [38;5;241m*[39m[38;5;241m*[39m({[38;5;124m"[39m[38;5;124mdata[39m[38;5;124m"[39m: data} [38;5;28;01mif[39;00m data [38;5;129;01mis[39;00m [38;5;129;01mnot[39;00m [38;5;28;01mNone[39;00m [38;5;28;01melse[39;00m {}),
[1;32m   3595[0m         [38;5;241m*[39m[38;5;241m*[39mkwargs,
[1;32m   3596[0m     )

File [0;32m/opt/anaconda3/envs/causal-inference-book/lib/python3.11/site-packages/matplotlib/axes/_axes.py:1724[0m, in [0;36mAxes.plot[0;34m(self, scalex, scaley, data, *args, **kwargs)[0m
[1;32m   1481[0m [38;5;250m[39m[38;5;124;03m"""[39;00m
[1;32m   1482[0m [38;5;124;03mPlot y versus x as lines and/or markers.[39;00m
[1;32m   1483[0m 
[0;32m   (...)[0m
[1;32m   1721[0m [38;5;124;03m(``'green'``) or hex strings (``'#008000'``).[39;00m
[1;32m   1722[0m [38;5;124;03m"""[39;00m
[1;32m   1723[0m kwargs [38;5;241m=[39m cbook[38;5;241m.[39mnormalize_kwargs(kwargs, mlines[38;5;241m.[39mLine2D)
[0;32m-> 1724[0m lines [38;5;241m=[39m [[38;5;241m*[39m[38;5;28mself[39m[38;5;241m.[39m_get_lines([38;5;28mself[39m, [38;5;241m*[39margs, data[38;5;241m=[39mdata, [38;5;241m*[39m[38;5;241m*[39mkwargs)]
[1;32m   1725[0m [38;5;28;01mfor[39;00m line [38;5;129;01min[39;00m lines:
[1;32m   1726[0m     [38;5;28mself[39m[38;5;241m.[39madd_line(line)

File [0;32m/opt/anaconda3/envs/causal-inference-book/lib/python3.11/site-packages/matplotlib/axes/_base.py:303[0m, in [0;36m_process_plot_var_args.__call__[0;34m(self, axes, data, *args, **kwargs)[0m
[1;32m    301[0m     this [38;5;241m+[39m[38;5;241m=[39m args[[38;5;241m0[39m],
[1;32m    302[0m     args [38;5;241m=[39m args[[38;5;241m1[39m:]
[0;32m--> 303[0m [38;5;28;01myield from[39;00m [38;5;28mself[39m[38;5;241m.[39m_plot_args(
[1;32m    304[0m     axes, this, kwargs, ambiguous_fmt_datakey[38;5;241m=[39mambiguous_fmt_datakey)

File [0;32m/opt/anaconda3/envs/causal-inference-book/lib/python3.11/site-packages/matplotlib/axes/_base.py:499[0m, in [0;36m_process_plot_var_args._plot_args[0;34m(self, axes, tup, kwargs, return_kwargs, ambiguous_fmt_datakey)[0m
[1;32m    496[0m     axes[38;5;241m.[39myaxis[38;5;241m.[39mupdate_units(y)
[1;32m    498[0m [38;5;28;01mif[39;00m x[38;5;241m.[39mshape[[38;5;241m0[39m] [38;5;241m!=[39m y[38;5;241m.[39mshape[[38;5;241m0[39m]:
[0;32m--> 499[0m     [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m([38;5;124mf[39m[38;5;124m"[39m[38;5;124mx and y must have same first dimension, but [39m[38;5;124m"[39m
[1;32m    500[0m                      [38;5;124mf[39m[38;5;124m"[39m[38;5;124mhave shapes [39m[38;5;132;01m{[39;00mx[38;5;241m.[39mshape[38;5;132;01m}[39;00m[38;5;124m and [39m[38;5;132;01m{[39;00my[38;5;241m.[39mshape[38;5;132;01m}[39;00m[38;5;124m"[39m)
[1;32m    501[0m [38;5;28;01mif[39;00m x[38;5;241m.[39mndim [38;5;241m>[39m [38;5;241m2[39m [38;5;129;01mor[39;00m y[38;5;241m.[39mndim [38;5;241m>[39m [38;5;241m2[39m:
[1;32m    502[0m     [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m([38;5;124mf[39m[38;5;124m"[39m[38;5;124mx and y can be no greater than 2D, but have [39m[38;5;124m"[39m
[1;32m    503[0m                      [38;5;124mf[39m[38;5;124m"[39m[38;5;124mshapes [39m[38;5;132;01m{[39;00mx[38;5;241m.[39mshape[38;5;132;01m}[39;00m[38;5;124m and [39m[38;5;132;01m{[39;00my[38;5;241m.[39mshape[38;5;132;01m}[39;00m[38;5;124m"[39m)

[0;31mValueError[0m: x and y must have same first dimension, but have shapes (100,) and (1000,)

